### PR TITLE
gh-106368: Increase coverage for Argument Clinic output directive

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-106368

### CODE REVIEW

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -249,6 +249,59 @@ def test_cpp_monitor_fail_no_matching_if(self):
...
+        out = self.clinic.parse(raw)
+        # The generated output will differ for every run, but we can check that
+        # it starts with the clinic block, we check that it contains all the
+        # expected fields, and we check that it contains the checksum line.
+        self.assertTrue(out.startswith(dedent("""
+            /*[clinic input]
+            output print 'I told you once.'
+            [clinic start generated code]*/
+        """)))
+        fields = {
```
Comment (sobolevn) [inline Lib/test/test_clinic.py:285]: Why `set`? :)
Since it is unordered it is not the best candidate for iteration in my opinion.

**Review by sobolevn — APPROVED** [review]
My question is not blocking :)
Thanks!

File: Lib/test/test_clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_clinic.py:285]: Order is not important, so a set is fine. I agree it is a little bit surprising, though. A good argument for changing it to a tuple, is that a tuple would not generate confusion for future contributors :)

Comment (ambv) [reply Lib/test/test_clinic.py:285]: Actually if the order shouldn't matter, a set is a good way to ensure it.

Comment (erlend-aasland) [reply Lib/test/test_clinic.py:285]: True. Also, I'm on mobile, so I won't change it now anyways :)

**Comment by erlend-aasland** [comment]
Thanks for the reviews!

**Comment by erlend-aasland** [comment]
I'll fix the backport later, unless someone beats me to it :)

**Pull request merged**